test/server/http_filter_base_test.cc

```diff
@@ -0,0 +1,89 @@
+#include "server/http_dynamic_delay_filter.h"
+#include "server/http_test_server_filter.h"
+#include "server/http_time_tracking_filter.h"
+
+#include "test/server/http_filter_integration_test_base.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+using namespace testing;
+
+class HttpFilterBaseIntegrationTest
+    : public HttpFilterIntegrationTestBase,
+      public testing::TestWithParam<
+          std::tuple<Envoy::Network::Address::IpVersion, absl::string_view, bool>> {
+public:
+  HttpFilterBaseIntegrationTest() : HttpFilterIntegrationTestBase(std::get<0>(GetParam())){};
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    IpVersions, HttpFilterBaseIntegrationTest,
+    ::testing::Combine(testing::ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()),
+                       testing::ValuesIn({absl::string_view(R"EOF(
+name: time-tracking
+typed_config:
+  "@type": type.googleapis.com/nighthawk.server.ResponseOptions
+  emit_previous_request_delta_in_response_header: "foo"
+)EOF"),
+                                          absl::string_view(R"EOF(
+name: dynamic-delay
+typed_config:
+  "@type": type.googleapis.com/nighthawk.server.ResponseOptions
+  static_delay: 0.1s
+)EOF"),
+                                          absl::string_view("name: test-server")}),
+                       testing::ValuesIn({true, false})));
+
+// Verify extensions are well-behaved when it comes to:
+// - GET requests.
+// - POST requests with an entity body (takes a slightly different code path).
+// - Valid but empty configuration.
+// - Bad request-level json configuration.
+// We will be low on functional expectations for the extensions, but this will catch hangs and
+// ensure that bad configuration handling is in-place.
+TEST_P(HttpFilterBaseIntegrationTest, BasicExtensionFlows) {
```

Comment:
Should we split these into separate tests? I'm not sure what the preference in the code base is, but it seems like each chunk is testing a different behaviour and might deserve a separate test?

Reply:
I can see your point, and in general I agree that the clearer test names that arise out of more finely grained separate tests are better. 
But in this case, when splitting this, up we would have to duplicate the parameter unpacking for the TEST_P params across the newly split up tests and/or add facilities to help with that. 
This was also sort of pre-existing (as we can eliminate three different instances of similar code in the test code for three extensions).

So personally I lean towards the compact version as is is now.  With the context above, are you comfortable with leaving this as-is?

Reply:
Yeah, makes sense to me.